stream.go

```diff
@@ -130,20 +170,25 @@ func (stream *Stream) stream(r io.ReadCloser) {
 		}
 		stream.Events <- ev
 	}
+}
+
+func (stream *Stream) retryRestartStream() {
 	backoff := stream.retry
 	for {
-		time.Sleep(backoff)
 		if stream.Logger != nil {
 			stream.Logger.Printf("Reconnecting in %0.4f secs\n", backoff.Seconds())
 		}
-
+		time.Sleep(backoff)
```

Comment:
I was also thinking about this, but I think this way it is "better". The reason is, once an error is encountered the error event is sent over the Errors channel and on the same time the retry loop will start. Consumers may want to close the stream, given they receive an Error on the channel. When I do now the time.Sleep before I check whether the channel is closed, I give the consumers the option to Close the stream in time, without the need that the additional retry request is sent. 

And I think it is no problem from a resource consumption point of view, to let the go route live a couple of seconds later. WDYT?